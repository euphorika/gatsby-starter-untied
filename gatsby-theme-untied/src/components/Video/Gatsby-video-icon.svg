<svg id="Ebene_1" data-name="Ebene 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 61.6 65.88"><defs><style>.cls-1{opacity:0.9;}.cls-2{fill:#fff;}</style></defs><title>Gatsby-video-icon</title><g class="cls-1"><path class="cls-2" d="M43.83,91.77s.25,11.83,10.62,6.11l45.3-25s10.36-5.72,0-11.52L52.84,35s-10.32-5.8-10.07,6Z" transform="translate(-42.77 -33.56)"/></g></svg>
