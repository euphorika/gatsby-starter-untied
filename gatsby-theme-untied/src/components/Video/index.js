import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.module.styl'

const Video = ({
  videos,
  poster,
  playsInline,
  autoPlay,
  preload,
  loop,
  muted,
  controls,
}) => (
  <section className={styles.videoContainer}>
    <div className={styles.innerContainer}>
      <video
        poster={poster}
        preload={preload}
        playsInline={playsInline}
        muted={muted}
        loop={loop}
        controls={controls}
        autoPlay={autoPlay}
      >
        {videos.map((video, id) => (
          <source key={video.id} src={video.src} type={video.type} />
        ))}
      </video>
    </div>
  </section>
)

Video.defaultProps = {
  preload: 'auto',
  playsInline: true,
  muted: true,
  loop: true,
  controls: true,
  autoPlay: true,
}

Video.propTypes = {
  poster: PropTypes.string,
  preload: PropTypes.oneOf(['none', 'preload', 'auto']),
  playsInline: PropTypes.bool,
  autoPlay: PropTypes.bool,
  muted: PropTypes.bool,
  loop: PropTypes.bool,
  controls: PropTypes.bool,
  src: PropTypes.string,
  type: PropTypes.string,
}

export default Video
