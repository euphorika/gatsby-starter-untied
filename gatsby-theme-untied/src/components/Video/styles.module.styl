@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.video-section
  margin 0
  width: 100%
  height: 100%
  text-align: center;

  .video-container
    width: 50%
    height: 100%
    display: flex
