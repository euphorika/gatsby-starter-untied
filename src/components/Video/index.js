import React from 'react'
import styles from './styles.module.styl'

import video from './elephants_dream.webm'

const Video = () => (
  <section className={styles.videoSection}>
    <video playsinline loop controls allowFullScreen>
      <source src={video} type="video/webm" />
    </video>
  </section>
)

export default Video
