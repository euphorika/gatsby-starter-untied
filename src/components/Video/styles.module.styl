@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.video-section
  margin 0
  width: 100%
  height: auto
  text-align: center;
