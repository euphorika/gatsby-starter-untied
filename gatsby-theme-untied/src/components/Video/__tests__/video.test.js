import React from 'react'
import renderer from 'react-test-renderer'
import Video from '../'

// required array parameter
const videos = [
  {
    node: {
      name: 'video 1',
      src: '#',
      type: 'type',
    },
  },
  {
    node: {
      name: 'video 1',
      src: '#',
      type: 'type',
    },
  },
]

describe('video', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<video entries={videos} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<video entries={videos} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
