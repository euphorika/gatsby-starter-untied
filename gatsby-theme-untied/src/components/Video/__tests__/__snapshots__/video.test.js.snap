// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video renders correctly 1`] = `
<video
  entries={
    Array [
      Object {
        "node": Object {
          "name": "video 1",
          "src": "#",
          "type": "type",
        },
      },
      Object {
        "node": Object {
          "name": "video 1",
          "src": "#",
          "type": "type",
        },
      },
    ]
  }
/>
`;

exports[`video renders correctly 2`] = `
<video
  entries={
    Array [
      Object {
        "node": Object {
          "name": "video 1",
          "src": "#",
          "type": "type",
        },
      },
      Object {
        "node": Object {
          "name": "video 1",
          "src": "#",
          "type": "type",
        },
      },
    ]
  }
/>
`;
